### PR TITLE
Feature: Add Http-Polling and Prometheus Components

### PR DESCRIPTION
**Pull request by n14s**
Addresses #34 

This PR contains two new updater components, their example modules and an auxiliary component:
- Http Polling Updater
- Prometheus Updater
- Http-jsonata-aas example
- Prometheus-jsonata-aas example
- Timer (auxiliary component)

With these components, data from Http or Prometheus Endpoints can be integrated into the AAS.
Especially integrating device data via tools like Prometheus is essential for creating digital twins via AAS.

### Technical details
- Prometheus component is an extension of the HttpPolling component, facilitating the query of metrics
- Route creation in UpdaterRouteBuilder has been extended to support the components
- HttpPolling example works out of the box, Prometheus example requires a basic Prometheus + nodeexporter instance running
- This PR only devolps its full functionality with #66, since its goal is to be deployed, while still being able to be configured.

### Try it out
The easiest way to try it out is to clone and run [this docker-compose stack](https://github.com/n14s/prometheus-aas-docker).\
It collects metric data and sends it to the AAS using the Prometheus Component.  

The core element of the compose stack is the Prometheus-AAS-Router service, where the Updater with the Prometheus component resides.\
[Check out its repo](https://github.com/n14s/prometheus-aas-router) for more details on how the the underlying Java application works and how the image is built

#### Camels Http component doesn't fit well into Basyx Updater route scheme:
Routes and route creation works different with camels Http component, as it starts with a timer

`from(timer).to(consumer).to(producer)`

instead of starting with the datasource

`from(consumer).to(producer)`

Therefore, conforming the Updater route scheme provides:\
Datasource = timer \
Datasink[0] = Http/Prometheus \
Datasink[1] = AAS  

This can be explored in the routes.json of the HttpPolling/Prometheus examples or in the demonstration compose stack.

In this PR the workaround results in a working solution, but the Basyx Updater scheme should consider altering its scheme for a cleaner integration of http endpoints.

### CODE REVIEW

File: basyx.components/basyx.components.updater/basyx.components.updater.core/src/main/java/basyx/components/updater/core/routebuilder/UpdaterRouteBuilder.java [hunk 1]
```diff
@@ -72,11 +69,20 @@ private void createRoute(RoutesConfiguration configuration, RouteConfiguration r
 		String[] dataSinkEndpoints = createDatasinkEndpoint(configuration, routeConfig.getDatasinks());
 		String[] dataTransformerEndpoints = createDataTransformerEndpoint(configuration, routeConfig.getTransformers());
 		String routeId = routeConfig.getRouteId();
+
 		if (Strings.isNullOrEmpty(routeConfig.getDelegator())) {
-			if (dataTransformerEndpoints == null || dataTransformerEndpoints.length == 0) {
-				from(dataSourceEndpoint).routeId(routeId).to("log:" + routeId).to(dataSinkEndpoints).to("log:" + routeId);
+			if (dataSourceEndpoint.startsWith("timer") == true) {
```
Comment (FrankSchnicke) [inline basyx.components/basyx.components.updater/basyx.components.updater.core/src/main/java/basyx/components/updater/core/routebuilder/UpdaterRouteBuilder.java:74]: Please remove the "== true"

**Review by FrankSchnicke — CHANGES_REQUESTED** [review]
Many thanks for this comprehensive PR! Would it be possible to address the single review point?

We will change the UpdaterRouteBuilder in future to support a more consistent configuration.

File: basyx.components/basyx.components.updater/basyx.components.updater.core/src/main/java/basyx/components/updater/core/routebuilder/UpdaterRouteBuilder.java (hunk 1 above)
Comment (n14s) [reply basyx.components/basyx.components.updater/basyx.components.updater.core/src/main/java/basyx/components/updater/core/routebuilder/UpdaterRouteBuilder.java:74]: Will do

**Comment by n14s** [comment]
Did 536d solve the remark?

**Review by FrankSchnicke — APPROVED** [review]

**Comment by FrankSchnicke** [comment]
Yes it did - I somehow missed this update. Thank you for pointing out the fix.

**Pull request merged**